src/overlays/actors/ovl_En_Horse/z_en_horse.c

```diff
@@ -90,253 +139,3541 @@ static ColliderJntSphInit D_80A65F28 = {
+    { -947, -1, -1604, 10, 0x4002, },
+};
+struct RaceInfo sIngoRace = { 8, sIngoRaceWaypoints };
+s32 sAnimSoundFrames[] = { 0, 16 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse/func_80A63FA8.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneScale, 600, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 300, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse/func_80A64150.s")
+u8 sResetNoInput[] = { 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse/func_80A6437C.s")
+s32 sIdleAnimIds[] = { 1, 3, 0, 3, 1, 0 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse/func_80A64480.s")
+s16 sIngoAnimations[] = { 7, 6, 2, 2, 1, 1, 0, 0, 0, 0 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse/func_80A64578.s")
+void EnHorse_CsMoveInit(EnHorse* this, GlobalContext* globalCtx, CsCmdActorAction* action);
+void EnHorse_CsJumpInit(EnHorse* this, GlobalContext* globalCtx, CsCmdActorAction* action);
+void EnHorse_CsRearingInit(EnHorse* this, GlobalContext* globalCtx, CsCmdActorAction* action);
+void EnHorse_WarpMoveInit(EnHorse* this, GlobalContext* globalCtx, CsCmdActorAction* action);
+void EnHorse_CsWarpRearingInit(EnHorse* this, GlobalContext* globalCtx, CsCmdActorAction* action);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse/func_80A6467C.s")
+EnHorseCsFunc sCutsceneInitFuncs[] = {
+    NULL, EnHorse_CsMoveInit, EnHorse_CsJumpInit, EnHorse_CsRearingInit, EnHorse_WarpMoveInit, EnHorse_CsWarpRearingInit,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse/EnHorse_Update.s")
+void EnHorse_CsMoveToPoint(EnHorse* this, GlobalContext* globalCtx, CsCmdActorAction* action);
+void EnHorse_CsJump(EnHorse* this, GlobalContext* globalCtx, CsCmdActorAction* action);
+void EnHorse_CsRearing(EnHorse* this, GlobalContext* globalCtx, CsCmdActorAction* action);
+void EnHorse_CsWarpMoveToPoint(EnHorse* this, GlobalContext* globalCtx, CsCmdActorAction* action);
+void EnHorse_CsWarpRearing(EnHorse* this, GlobalContext* globalCtx, CsCmdActorAction* action);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse/func_80A64ED4.s")
+EnHorseCsFunc sCutsceneActionFuncs[] = {
+    NULL, EnHorse_CsMoveToPoint, EnHorse_CsJump, EnHorse_CsRearing, EnHorse_CsWarpMoveToPoint, EnHorse_CsWarpRearing,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse/func_80A64F14.s")
+typedef struct {
+    s32 csAction;
+    s32 csFuncIdx;
+} csActionEntry;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse/func_80A65014.s")
+csActionEntry sCsActionTable[] = {
+    { 36, 1, }, { 37, 2, },
+    { 38, 3, }, { 64, 4, },
+    { 65, 5, },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse/func_80A6506C.s")
+struct RaceWaypoint sHbaWaypoints[] = {
+    { 3600, 1413, -5055, 11, 0x8001, },
```

Comment:
relevant throughout the file but we dont generally put `struct` when declaring new data